openapi3/schema_formats_test.go

```diff
@@ -98,10 +91,319 @@ components:
 
 	err = doc.Components.Schemas["Something"].Value.VisitJSON(map[string]interface{}{
 		`ip`: `123.0.0.11111`,
-	})
+	}, SetOpenAPIMinorVersion(1))
 
-	require.EqualError(t, err, `Error at "/ip": Not an IP address`)
+	// assert, do not require to ensure SchemaErrorDetailsDisabled can be set to false
+	assert.ErrorContains(t, err, `Error at "/ip"`)
 
-	delete(SchemaStringFormats, "ipv4")
 	SchemaErrorDetailsDisabled = false
 }
+
+func TestStringFormatsStartingWithOpenAPIMinorVersion(t *testing.T) {
+	DefineStringFormatStartingWithOpenAPIMinorVersion("test", 0, "test0", false)
+	for i := uint64(0); i < 10; i++ {
+		if assert.Contains(t, SchemaStringFormats, "test") &&
```

Comment:
Please prefer the `require` package over `assert`. This makes for fewer nesting.